pytorch_lightning/trainer/distrib_data_parallel.py

```diff
@@ -165,6 +165,10 @@ class TrainerDDPMixin(ABC):
     num_nodes: int
     node_rank: int
 
+    @property
+    def is_global_zero(self) -> int:
```

Comment:
I think we should avoid these abstract functions if possible, they make it hard to understand where the actual function is implemented. Can we just store the global_zero value in trainer __init__?

Reply:
yeah, i also hate these abstract functions... @Borda any way to get rid of these?

They happen for anything in a mixin where the function isn't defined there, it's not because it's a property.

Why don't we use another PR to get rid of all of these correctly? i agree it's frustrating to click inspect and be taken to an abstract function

Reply:
Get rid of those means redo all mixins...

Reply:
sooner or later the mixins will have to get redone anyway out of necessity. the current misuse just hides the fact that Trainer is a god class (pssst, not too loud!) :)